lib/gtfs/hastus/trip.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Gtfs.Hastus.Trip do
+  alias Gtfs.{Block, Csv, Route, Run, Trip}
+  alias Gtfs.Hastus.{Place, Schedule}
+
+  @type t :: %__MODULE__{
+          schedule_id: Schedule.id(),
+          run_id: Run.id(),
+          block_id: Block.id(),
+          start_time: String.t(),
+          end_time: String.t(),
+          start_place: Place.id(),
+          end_place: Place.id(),
+          # nil means nonrevenue
+          route_id: Route.id() | nil,
+          trip_id: Trip.id()
+        }
+
+  @enforce_keys [
+    :schedule_id,
+    :run_id,
+    :block_id,
+    :start_time,
+    :end_time,
+    :start_place,
+    :end_place,
+    :trip_id
+  ]
+
+  defstruct [
+    :schedule_id,
+    :run_id,
+    :block_id,
+    :start_time,
+    :end_time,
+    :start_place,
+    :end_place,
+    :route_id,
+    :trip_id
+  ]
+
+  @spec from_csv_row(Csv.row()) :: t()
+  def from_csv_row(row) do
+    %__MODULE__{
+      schedule_id: row["schedule_id"],
+      run_id: run_id(row),
+      block_id: String.replace(row["block_id"], " ", ""),
```

Comment:
This `String.replace` function is repeated a couple of times. Could it be refactored out?

Reply:
It's just once in each file, and it's a pretty simple library call, so pulling it to a separate file is probably overkill.